src/util.js

```diff
@@ -107,3 +107,80 @@ export function format_memory_and_limit(usage, limit) {
+                            varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id})
+                                    .then(reply => {
+                                        resolve(JSON.parse(reply.container));
+                                    })
+                                    .catch(ex => reject(new Error("Failed to do InspectContainer call:", ex, JSON.stringify(ex))));
+                        });
+                    });
+                })
+                .then(function() {
+                    const containerCount = newContainersMeta.length;
+                    Promise.all(inspectRet)
+                            .then((inspectRet) => {
+                                let runEle = newContainersMeta.filter((ele) => {
+                                    return ele.status === "running";
+                                });
+                                inspectRet.map((inspectRet) => {
+                                    newContainers[inspectRet.ID] = inspectRet;
+                                });
+                                if (runEle.length === 0 && Object.keys(newContainers).length === containerCount) {
+                                    resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                                }
+                            })
+                            .catch(ex => console.error("Failed to do InspectContainer call:", ex, JSON.stringify(ex)));
+                })
+
+                .then(function() {
+                    containerStatsRet = newContainersMeta.filter((ele) => {
+                        return ele.status === "running";
+                    }).map((container) => {
+                        return new Promise((resolve, reject) => {
+                            varlinkCall(PODMAN, "io.podman.GetContainerStats", {name: container.id})
+                                    .then(reply => resolve({ctrId: container.id, ctrStats:reply.container}))
+
+                                    .catch(ex => {
+                                        console.error("Failed to do GetContainerStats call:", ex, JSON.stringify(ex));
+                                        reject(new Error("Failed to do GetContainerStats call:", ex, JSON.stringify(ex)));
+                                    });
+                        });
+                    });
+                    countRun = containerStatsRet.length;
+                })
+                .then(() => {
+                    Promise.all(containerStatsRet)
+                            .then((containerStatsRet) => {
+                                containerStatsRet.map((containerStatsRet) => {
+                                    newContainersStats[containerStatsRet.ctrId] = containerStatsRet.ctrStats;
+                                });
+                                if (countRun > 0 && Object.keys(newContainersStats).length === countRun) {
```

Comment:
Same story for the length comparison (should just go).

So the `countRun` here would make sure to not resolve the returned promise again if it already happened above. But all of this code just works for no running containers -- `Promise.all([]).then()` immediately resolves and runs the then handler.

So do all of the work above unconditionally, and unconditionally resolve the returned promise here, and do it all in one .then handler. The code will be much smaller, simpler, and avoid branches.